### PR TITLE
wine{unstable,staging}: 7.2 -> 7.4,vkd3d: 1.2 -> 1.3

### PR DESCRIPTION
**Pull request by bhankas**
###### Description of changes

Wine updates since 7.2:
https://www.winehq.org/announce/7.4
https://www.winehq.org/announce/7.3
https://www.winehq.org/announce/7.2 (also contains announcement for wine-mono 7.1.1)

vkd3d:
https://www.winehq.org/news/2022030201

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bhankas** [comment]
staging package builds are failing with this error:

```
Applying /build/wine-7.4/patches/api-ms-win-Stub_DLLs/0009-ext-ms-win-xaml-pal-l1-1-0-Add-dll-and-add-stub-for-.patch
patching configure.ac
1 out of 1 hunk FAILED -- saving rejects to file configure.ac.rej
[PATCH] ERR: Textual patch did not apply, aborting.
ERROR: Failed to apply patch, aborting!
```

I have tried looking more into it, but since this is my first day compiling wine or looking at its nixpkgs derivations, I am not able to make much sense on how to proceed on this. `nixpkgs-review` is still running, so I will let it finish before trying any further.

Also for some reason trying to convert this PR into draft returns 404.

**Comment by bhankas** [comment]
I had to stop `nixpkgs-review` after 5 hours due to other reasons. If someone wants to try building this PR, or even build on top of this, please go ahead.

**Comment by SFrijters** [comment]
Not at a PC so I can't check this but some remarks:
- You need to update the hash for staging as well; in the current state of the PR it will probably use cached 7.2 staging patches against 7.4 unstable wine.
- The mono version was already updated to 7.1.1 for unstable/staging, the mono version for stable should probably be kept as-is.

**Comment by bhankas** [comment]
Thanks for taking a look @SFrijters.  Changed according to your suggestions. Trying to build one more time on my machine, but not hoping for miracle.

**Comment by bhankas** [comment]
`nixpkgs-review` is still going on my system, but there are no more patch related failures, so I am hopeful.

I looks like `airwave` build will fail due to unrelated reason, however:

```
error: builder for '/nix/store/yhv15i75zp39qyfzrf4c03fg8h82h1cr-vstsdk368_08_11_2017_build_121.zip.drv' failed with exit code 1;
       last 10 log lines:
       > ***
       > Unfortunately, we cannot download file vstsdk368_08_11_2017_build_121.zip automatically.
       > Please go to http://www.steinberg.net/en/company/developers.html to download it yourself, and add it to the Nix store
       > using either
       >   nix-store --add-fixed sha256 vstsdk368_08_11_2017_build_121.zip
       > or
       >   nix-prefetch-url --type sha256 file:///path/to/vstsdk368_08_11_2017_build_121.zip
       >
       > ***
       >
       For full logs, run 'nix log /nix/store/yhv15i75zp39qyfzrf4c03fg8h82h1cr-vstsdk368_08_11_2017_build_121.zip.drv'.
error: 1 dependencies of derivation '/nix/store/mvywln4rzk6kvzaidd9i58x0nv8hyyyj-vstsdk368_08_11_2017_build_121.drv' failed to build
```

**Comment by SFrijters** [comment]
Yeah, airwave is nonfree and the required source zip is no longer available last I checked.
But I think it uses stable so if you don't bump mono nothing should change?

**Comment by bhankas** [comment]
Thanks. After 8+ hours this thing is still not finished and will take a while more (I need a new machine, don't I?). I'm gonna let it finish now, sunk cost and all. But have no idea how to actually use it in a flake based config to override just wine and steam-run.

**Comment by bhankas** [comment]
After measly 12+ hours of compilation, this PR is now ready for review. @SFrijters kindly do the honors.

Packages built:
- grapejuice
- lutris
- lutris-free
- lutris-unwrapped
- pipelight
- playonlinux
- q4wine
- winbox
- wine
- wine-staging
- wineWayland
- wine64
- wine64Packages.staging
- wine64Packages.stagingFull
- wine64Packages.unstable
- wine64Packages.unstableFull
- wine64Packages.waylandFull
- winePackages.staging
- winePackages.unstable
- winePackages.unstableFull
- wineWowPackages.full
- wineWowPackages.staging
- wineWowPackages.stagingFull
- wineWowPackages.unstable
- wineWowPackages.unstableFull
- wineWowPackages.waylandFull
- yabridge
- yabridgectl

`airwave` failed to built as predicted due to previously stated reason.

File: pkgs/applications/emulators/wine/vkd3d.nix [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+  version = "1.3";
 
   src = fetchurl {
     url = "https://dl.winehq.org/vkd3d/source/vkd3d-${version}.tar.xz";
-    sha256 = "0szr1lw3xbgi9qjm13d1q4gyzzwv8i5wfxiwjg6dmwphrc7h6jxh";
+    sha256 = "134b347806d34a4d2b39ea29ff1c2b38443793803a3adc50800855bb929fb8b2";
   };
 
-  buildInputs = [ vulkan-headers spirv-headers ]
+  buildInputs = [ vulkan-headers spirv-headers flex bison ]
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/wine/vkd3d.nix:14]:
```suggestion
  nativeBuildInputs = [ flex bison ]

  buildInputs = [ vulkan-headers spirv-headers ]
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request merged**